vector-im/element-x-android
# Add Session Verification flow

## Changes

Adds a self-verification screen when the user is not verified. Implements #89 .

## Tests

1. Sign in with any account.
2. Tap on 'Continue' in the verify session prompt shown in the room list.
3. Request verification, accept it on another device.
4. Either complete the verification, cancel it or make it fail at different stages to try to find corner cases.
5. Once verified, make sure the 'Verification successful' message is displayed only once and the verify session prompt is gone from the room list.

## Screenshots

| Light mode | Dark mode |
|-|-|
|![image](https://user-images.githubusercontent.com/480955/224948206-c5a3d41e-052a-4421-b122-7bcc61564a1a.png)|![image](https://user-images.githubusercontent.com/480955/224948492-f743df38-20e9-4c30-ad82-0d9159a46e63.png)|
|![image](https://user-images.githubusercontent.com/480955/224948716-2868de41-0663-4618-ad57-7b510e11dcd0.png)|![image](https://user-images.githubusercontent.com/480955/224949123-84fdaa23-761f-4cb1-90bd-2ad06ad07946.png)|
|![image](https://user-images.githubusercontent.com/480955/224948887-51495b27-fb3f-45bf-a56b-2526502700f2.png)|![image](https://user-images.githubusercontent.com/480955/224948939-45902df1-233c-43f7-bc2b-b1e850124243.png)|
|![Screenshot_1678784501](https://user-images.githubusercontent.com/480955/224949761-bc044ebf-bbe1-48ef-b2cb-9ed06e2ecde0.png)|![Screenshot_1678784496](https://user-images.githubusercontent.com/480955/224949816-966c6e0e-efff-4def-8789-f5b26badae61.png)|
|![image](https://user-images.githubusercontent.com/480955/224949366-1d6663e0-b00a-40b4-862d-fd69d10d8168.png)|![image](https://user-images.githubusercontent.com/480955/224949430-c2482aaf-a44a-48ff-bf52-aca615c9deed.png)|
|![Screenshot_1678784511](https://user-images.githubusercontent.com/480955/224949996-9571fb15-01e1-479e-be13-c4a21dbf0b0e.png)|![Screenshot_1678784514](https://user-images.githubusercontent.com/480955/224950027-a0ccd9ef-6d0f-458e-a866-612f0e7a68bd.png)|
